pkg/actors/actors.go

```diff
@@ -1106,13 +1126,26 @@ func (a *actorsRuntime) CreateTimer(ctx context.Context, req *CreateTimerRequest
 					if err != nil {
 						log.Errorf("error invoking timer on actor %s: %s", actorKey, err)
 					}
-				} else {
+					if req.Repeats > 0 {
+						cnt++
+					}
+				}
+				if !exists || (req.Repeats > 0 && cnt >= req.Repeats) {
+					if duration > 0 {
+						stopTimer.Stop()
+					}
 					a.DeleteTimer(ctx, &DeleteTimerRequest{
 						Name:      req.Name,
 						ActorID:   req.ActorID,
 						ActorType: req.ActorType,
 					})
 				}
+			case <-stopTimerC:
```

Comment:
Can we just handle the duration together with the repeat count?

Reply:
no, separate channels